### PR TITLE
added controllers, mappers and DTOs for entities

### PR DESCRIPTION
**Pull request by BusyhinD**

### CODE REVIEW

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import mate.academy.spring.service.ShoppingCartService;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@CrossOrigin(origins = "*")
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/OrderController.java:22]: No need for this annotation

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        return orderService.getOrdersHistory(user)
+                .stream()
+                .map(orderDtoResponseMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto add(@RequestParam Long userId) {
+        User user = new User();
+        user.setId(userId);
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/OrderController.java:55]: It is better to get the user from the service userService.get(userId)

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.orderDtoResponseMapper = orderDtoResponseMapper;
+        this.shoppingCartDtoRequestMapper = shoppingCartDtoRequestMapper;
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getHistoryForUser(@RequestParam Long userId) {
+        User user = new User();
+        user.setId(userId);
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/OrderController.java:45]: the same

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                                          movieSessionDtoRequestMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.shoppingCartDtoResponseMapper = shoppingCartDtoResponseMapper;
+        this.movieSessionDtoRequestMapper = movieSessionDtoRequestMapper;
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
+        User user = new User();
+        user.setId(userId);
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/ShoppingCartController.java:42]: same

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    @PutMapping("/movie-sessions")
+    public ShoppingCartResponseDto update(
+            @RequestParam Long userId,
+            @RequestParam Long movieSessionId,
+            @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = movieSessionDtoRequestMapper.fromDto(movieSessionRequestDto);
+        movieSession.setId(movieSessionId);
+        User user = new User();
+        user.setId(userId);
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/ShoppingCartController.java:54]: same for movie session

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        return shoppingCartDtoResponseMapper.toDto(shoppingCartService.getByUser(user));
+    }
+
+    @PutMapping("/movie-sessions")
+    public ShoppingCartResponseDto update(
+            @RequestParam Long userId,
+            @RequestParam Long movieSessionId,
+            @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/ShoppingCartController.java:50]:
```suggestion
            ) {
```

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final DtoResponseMapper<OrderResponseDto, Order> orderDtoResponseMapper;
+    private final DtoRequestMapper<ShoppingCartRequestDto, ShoppingCart>
+            shoppingCartDtoRequestMapper;
```
Comment (uliana-andrushchuk) [inline src/main/java/mate/academy/spring/controller/OrderController.java:29]: This field is redundant.
I do not see usage of this `shoppingCartDtoRequestMapper` in this class

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]
Good job

**Pull request closed without merging**